_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
 permalink: /docs/testing-with-xquery/
 ---
 
-# Testing With XQuery
+## Testing XQuery With XQuery
 
-*TODO*
+### Scenario 1 - Test assert all exist with assert-equal for success
+```
+xquery version "1.0-ml";
 
-## Testing XQuery With XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
 
-*TODO*
+let $count := 2
+let $testSequence := (1,2)
+let $result := test:assert-all-exist($count, $testSequence)
+return test:assert-equal($result/@type/string(), "success")
+```
+### Scenario 2 - Test assert all exist with assert-equal for exception error
+```
+xquery version "1.0-ml";
 
-## Test Server-side JavaScript with XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $count := 3
+let $testSequence := (1,2)
+let $errorMessage := "Assert All Exist failed"
+let $result := 
+try{
+  test:assert-all-exist($count, $testSequence)
+} catch($err) {
+  $err
+}
+return test:assert-equal($result//error:code/string(), $errorMessage)
```

Comment:
I don't think we need this scenario.  I don't believe users of marklogic-unit-test will intentionally write tests that fail, so there isn't a need to show examples of how to catch those failures and guarantee they fail.